consensus/service/src/client_api_service.rs

```diff
@@ -165,26 +165,15 @@ mod client_api_tests {
     };
     use mc_util_grpc::auth::{AnonymousAuthenticator, TokenAuthenticator};
     use serial_test_derive::serial;
-    use std::{
-        sync::{
-            atomic::{AtomicUsize, Ordering::SeqCst},
-            Arc,
-        },
-        time::Duration,
-    };
-
-    fn get_free_port() -> u16 {
-        static PORT_NR: AtomicUsize = AtomicUsize::new(0);
-        PORT_NR.fetch_add(1, SeqCst) as u16 + 30100
-    }
+    use std::{sync::Arc, time::Duration};
 
     /// Starts the service on localhost and connects a client to it.
     fn get_client_server(instance: ClientApiService) -> (ConsensusClientApiClient, Server) {
         let service = consensus_client_grpc::create_consensus_client_api(instance);
         let env = Arc::new(Environment::new(1));
         let mut server = ServerBuilder::new(env.clone())
             .register_service(service)
-            .bind("127.0.0.1", get_free_port())
+            .bind("127.0.0.1", 0)
```

Comment:
I ran into some intermittent unit test failures on CI that suggested that client_api_service and peer_api_service unit tests were interfering with each other. I didn't dig too deeply into it, but this seems to fix things.

Reply:
it seems like a better fix would be to have all of the tests share a common `get_free_port` fn that is imported from a test utility crate